crates/starknet/src/starknet/mod.rs

```diff
@@ -159,6 +175,95 @@ impl Starknet {
 
         this.restart_pending_block()?;
 
+        // Re-execute transactions
+        if transactions.is_some() {
+            // is_some() is need here?
+            for (_hash, transaction) in transactions.unwrap_or_default().iter() {
```

Comment:
In case _hash isn't used 

```suggestion
            for (_, transaction) in transactions.unwrap_or_default().iter() {
```

Reply:
fixed in 6eae5a26d2a3fed5d02bcbab038cc9c232d227b3